pkg/downloader/manager.go

```diff
@@ -529,6 +530,14 @@ func (m *Manager) ensureMissingRepos(repoNames map[string]string, deps []*chart.
 
 		repoNames[dd.Name] = rn
 
+		// If repository is already present don't add to array. This will skip
+		// unnecessary index file downloading improving performance.
+		for _, item := range ru {
+			if item.URL == dd.Repository {
+				continue Outer
+			}
+		}
+
```

Comment:
Will this change the behavior? I would imagine if I had a workflow that expected `helm dependency update` to grab the latest index.yaml every time, this would break that, would it not?

```
dependencies:
  - name: console
    condition: console.enabled
    version: ">=0.5 <1.0"
    repository: https://charts.redpanda.com
```

This change would keep that version constraint from updating the subchart if a new `console` chart was released, right?

Reply:
Hi @joejulian 
This if statement only prevents the same repo being added multiple times to the array of repos to be updated as part of the dependency update, and therefore prevents the same repo being downloaded multiple times as part of the same dependency update. I believe the situation you mention will be handled as expected with the latest index.yaml being downloaded (but only once, irrespective of how many times it is referenced in the dependencies)